### PR TITLE
Fix invalid cast of QgsProperty as bool to store in strings

### PR DESCRIPTION
**Pull request by m-kuhn**
Followup https://github.com/qgis/QGIS/pull/42456

### CODE REVIEW

File: src/core/vectortile/qgsmapboxglstyleconverter.cpp [hunk 1]
```diff
@@ -1481,7 +1481,7 @@ void QgsMapBoxGlStyleConverter::parseSymbolLayer( const QVariantMap &jsonLayer,
             labelSettings.distUnits = context.targetUnit();
             labelSettings.dist = std::abs( textOffset.y() ) - textSize;
             labelSettings.lineSettings().setPlacementFlags( textOffset.y() > 0.0 ? QgsLabeling::BelowLine : QgsLabeling::AboveLine );
-            if ( textSizeProperty && textOffsetProperty.isEmpty() )
+            if ( !textSizeProperty )
```
Comment (nyalldawson) [inline src/core/vectortile/qgsmapboxglstyleconverter.cpp:1484]: that's not right...

Comment (m-kuhn) [reply src/core/vectortile/qgsmapboxglstyleconverter.cpp:1484]:
```suggestion
            if ( textSizeProperty && !textOffsetProperty )
```

better :see_no_evil:

Comment (m-kuhn) [reply src/core/vectortile/qgsmapboxglstyleconverter.cpp:1484]:
```suggestion
            if ( textSizeProperty && !textOffsetProperty )
```

better :see_no_evil:

Comment (nyalldawson) [reply src/core/vectortile/qgsmapboxglstyleconverter.cpp:1484]: yep

**Pull request merged**